File: test/jvm/backtype/storm/utils/UtilsTest.java
package backtype.storm.utils;

import org.junit.Test;

import java.util.List;
import java.util.ArrayList;
import java.util.Map;
import java.util.HashMap;

import static org.junit.Assert.assertTrue;
import static org.mockito.Mockito.mock;

import com.netflix.curator.CuratorZookeeperClient;
import com.netflix.curator.framework.CuratorFramework;
import com.netflix.curator.retry.ExponentialBackoffRetry;

import backtype.storm.Config;

public class UtilsTest {

  @Test
  public void testNewCuratorUsesExponentialBackoff() {
    @SuppressWarnings("unchecked")
    Map<String,Object> conf = (Map<String,Object>)Utils.readDefaultConfig();
    List<String> servers = new ArrayList<String>();
    servers.add("bogus_server");
    Object port = new Integer(42);
    CuratorFramework cf = Utils.newCurator(conf, servers, port);

    assertTrue(cf.getZookeeperClient().getRetryPolicy() instanceof ExponentialBackoffRetry);
  }
}

File: test/clj/backtype/storm/utils_test.clj
(ns backtype.storm.utils-test
  (:import [backtype.storm Config])
  (:import [backtype.storm.utils NimbusClient Utils])
  (:import [com.netflix.curator.retry ExponentialBackoffRetry])
  (:import [org.apache.thrift7.transport TTransportException])
  (:use [backtype.storm config util])
  (:use [clojure test])
)

(deftest test-new-curator-uses-exponential-backoff
  (let [expected_interval 2400
        expected_retries 10
        conf (merge (read-storm-config)
          {Config/STORM_ZOOKEEPER_RETRY_INTERVAL expected_interval
           Config/STORM_ZOOKEEPER_RETRY_TIMES expected_retries})
        servers ["bogus_server"]
        arbitrary_port 42
        curator (Utils/newCurator conf servers arbitrary_port)
        retry (-> curator .getZookeeperClient .getRetryPolicy)
       ]
    (is (.isAssignableFrom ExponentialBackoffRetry (.getClass retry)))
    (is (= (.getBaseSleepTimeMs retry) expected_interval))
    (is (= (.getN retry) expected_retries))
  )
)

(deftest test-getConfiguredClient-throws-RunTimeException-on-bad-config
  (let [storm-conf (merge (read-storm-config)
                     {STORM-THRIFT-TRANSPORT-PLUGIN
                       "backtype.storm.security.auth.SimpleTransportPlugin"
                      Config/NIMBUS_HOST ""
                      Config/NIMBUS_THRIFT_PORT 65535
                     })]
    (is (thrown? RuntimeException
      (NimbusClient/getConfiguredClient storm-conf)))
  )
)

(deftest test-getConfiguredClient-throws-RunTimeException-on-bad-args
  (let [storm-conf (read-storm-config)]
    (is (thrown? TTransportException
      (NimbusClient. storm-conf "" 65535)
    ))
  )
)
